README.md

```diff
@@ -128,17 +148,222 @@ To start Parity Ethereum as a regular user using `systemd` init:
+    ```
+  * Parity Ethereum (EthCore) Blockchain Synchronization
+    ```bash
+    ethcore-sync
+    ```
+  * Parity Ethereum Common Types
+    ```bash
+    common-types
+    ```
+  * Parity Ethereum Virtual Machines (VM) Support Library
+    ```bash
+    vm
+    ```
+  * Parity Ethereum WASM Interpreter
+    ```bash
+    wasm
+    ```
+  * Parity Ethereum WASM Test Run
+    ```bash
+    pwasm-run-test
+    ```
+  * Parity EVM Implementation
+    ```bash
+    evmbin
+    ```
+  * Parity Ethereum IPFS-compatible API
+    ```bash
+    parity-ipfs-api
+    ```
+  * Parity Ethereum JSON Deserialization
+    ```bash
+    ethjson
+    ```
+  * Parity Ethereum State Machine Generalization for Consensus Engines
+    ```bash
+    parity-machine
+    ```
+* Parity Ethereum (EthCore) Miner Interface
+  ```bash
+  ethcore-miner parity-local-store price-info ethcore-stratum using_queue
+  ```
+* Parity Ethereum (EthCore) Logger Implementation
+  ```bash
+  ethcore-logger
+  ```
+* Parity Ethereum Client C-Bindings Library
+  ```bash
+  parity-clib
+  ```
+* Parity Ethereum JSON-RPC Servers
+  ```bash
+  parity-rpc
+  ```
+* Parity Ethereum (EthCore) Secret Store
+  ```bash
+  ethcore-secretstore
+  ```
+* Parity Updater Service
+  ```bash
+  parity-updater parity-hash-fetch
+  ```
+* Parity Utilities
+  ```bash
+  ethcore-bloom-journal blooms-db dir eip-712 fake-fetch fastmap fetch ethcore-io
+  journaldb keccak-hasher len-caching-lock macros memory-cache memzero
+  migration-rocksdb ethcore-network ethcore-network-devp2p panic_hook
+  patricia-trie-ethereum registrar rlp_compress rlp_derive parity-runtime stats
+  time-utils triehash-ethereum unexpected parity-version
+  ```
+* Parity Whisper Protocol Implementation
+  ```bash
+  parity-whisper whisper-cli
+  ```
+
+### Contributing to documentation for Parity Ethereum packages
+
+[Document source code](https://doc.rust-lang.org/1.9.0/book/documentation.html) for Parity Ethereum packages by annotating the source code with documentation comments.
+
+Example (generic):
+```markdown
+/// Summary
+///
+/// Description
+///
+/// # Panics
+///
+/// # Errors
+///
+/// # Safety
+///
+/// # Examples
+///
+/// Summary of Example 1
+///
+/// ```rust
+/// // insert example 1 code here
+/// ```
+///
+```
+
+* Important notes:
+  * Documentation comments must use annotations with a triple slash `///`
+  * Modules are documented using `//!`
+```
+//! Summary (of module)
+//!
+//! Description (of module)
+```
+* Special section header is indicated with a hash `#`.
+  * `Panics` section requires an explanation if the function triggers a panic
+  * `Errors` section is for describing conditions under which a function of method returns `Err(E)` if it returns a `Result<T, E>`
+  * `Safety` section requires an explanation if the function is `unsafe`
+  * `Examples` section includes examples of using the function or method
+* Code block annotations for examples are included between triple graves, as shown above.
```

Comment:
Let's not teach the reader Rust, that is assumed.

Reply:
I've addressed this in commit 39cdd6f15ac1c27cc1b3f4f045c30a60c585c7da